### PR TITLE
Fix #249 - Add WebExtension theme integration setting

### LINKED ISSUE #249: Integrate with current theme
From 58 onwards, you can get the current theme properties: 

https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/theme/getCurrent

https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/theme/onUpdated

Why is this useful ?
Integrating with dynamic themes like [VivaldiFox](https://addons.mozilla.org/en-US/firefox/addon/vivaldifox/) or [Container Theme](https://addons.mozilla.org/en-US/firefox/addon/containers-theme/)

A lot of people have expressed their wish to see VivaldiFox work with their favourite sidebar tabs add-on.

### PR DESCRIPTION
**Pull request by nt1m**

### CODE REVIEW

**Comment by nt1m** [comment]
@eoger Could you please review this? Thanks :)

File: src/sidebar/lib/contextmenu/contextmenu.js [hunk 1]
```diff
@@ -45,11 +45,11 @@ ContextMenu.prototype = {
     const winHeight = window.innerHeight;
 
     const top = this.clickPosY + menuHeight > winHeight ?
-                (winHeight - menuHeight) :
-                this.clickPosY;
+      (winHeight - menuHeight) :
```
Comment (eoger) [inline src/sidebar/lib/contextmenu/contextmenu.js]: Please keep the indentation (here and other files).

File: src/sidebar/tabcenter.js [hunk 2]
```diff
@@ -14,11 +14,11 @@ TabCenter.prototype = {
 
     this._setupListeners();
 
+    const {id: windowId} = await windowPromise;
```
Comment (eoger) [inline src/sidebar/tabcenter.js]: Is there a reason behind this change?

File: src/sidebar/tabcenter.js [hunk 3]
```diff
@@ -69,12 +74,24 @@ TabCenter.prototype = {
       document.body.classList.remove("dark-theme");
     }
   },
+  set _themeIntegration(enabled) {
+    if (!enabled) {
+      this._applyTheme({});
```
Comment (eoger) [inline src/sidebar/tabcenter.js]: Please create a helper function `_resetTheme` that hides this implementation detail.

File: src/sidebar/tabcenter.js [hunk 4]
```diff
@@ -84,6 +101,29 @@ TabCenter.prototype = {
     if (prefs.hasOwnProperty("darkTheme")) {
       this._darkTheme = prefs.darkTheme;
     }
+    if (prefs.hasOwnProperty("themeIntegration")) {
+      this._themeIntegration = prefs.themeIntegration;
+    }
+  },
+  _applyTheme(theme) {
+    let setVariable = (cssVar, themeProps) => {
```
Comment (eoger) [inline src/sidebar/tabcenter.js]: `const`

File: src/sidebar/tabcenter.js [hunk 5]
```diff
@@ -84,6 +101,29 @@ TabCenter.prototype = {
...
+      for (let prop of themeProps) {
```
Comment (eoger) [inline src/sidebar/tabcenter.js]: `const prop`

File: src/sidebar/tabcenter.css [hunk 6]
```diff
@@ -1,14 +1,14 @@
 @import url("lib/contextmenu/contextmenu.css");
 
 :root {
-    --tab-background-normal: 0, 0%, 99%;
-    --tab-background-pinned: 0, 0%, 97%;
-    --tab-background-active: 0, 0%, 87%;
-    --tab-background-hover: 0, 0%, 91%;
+    --tab-background-normal: hsl(0, 0%, 99%);
+    --tab-background-pinned: hsla(0,0%,80%,.15);
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Space between `,` and `.`

File: src/sidebar/tabcenter.css [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+    --tab-background-active: hsl(0, 0%, 87%);
+    --tab-background-hover: hsla(0,0%,80%,.25);
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Same

File: src/sidebar/tabcenter.css [hunk 8]
```diff
@@ -643,43 +645,19 @@ body[platform="win"] #searchbox-input:placeholder-shown {
...
-}
-
-.tab:not(.active):hover > .tab-title-wrapper::after {
-    --tab-background: var(--tab-background-hover);
-}
-
-.tab:hover > .tab-title-wrapper::after {
-    transform: translateX(0);
+.tab-title-wrapper {
+    mask-image: linear-gradient(to left, transparent 0, black 30%);
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: I find 30% to be a bit too wide, how about we use the `2em` Firefox uses natively?

File: src/sidebar/tabcenter.css [hunk 9]
```diff
@@ -630,7 +632,7 @@ body[platform="win"] #searchbox-input:placeholder-shown {
 }
 
 .tab.discarded .tab-title {
-    color: var(--secondary-text-color);
+    opacity: 0.5;
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Note (unrelated): We should try to do something similar (`.7`) for the title wrapper when a tab is not active.

File: src/sidebar/tabcenter.css [hunk 10]
```diff
@@ -643,43 +645,19 @@ body[platform="win"] #searchbox-input:placeholder-shown {
...
-    --tab-background: var(--tab-background-active);
-}
-
-.tab:not(.active):hover > .tab-title-wrapper::after {
-    --tab-background: var(--tab-background-hover);
-}
-
-.tab:hover > .tab-title-wrapper::after {
-    transform: translateX(0);
+.tab-title-wrapper {
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: 🎉 These changes look great, that absolute pseudo-element was bothering me. It would have been nice to extract these changes in a different commit.

File: src/sidebar/tabcenter.js (hunk 2 above)
Comment (nt1m) [reply src/sidebar/tabcenter.js]: set _themeIntegration needs the windowId.

File: src/sidebar/lib/contextmenu/contextmenu.js (hunk 1 above)
Comment (nt1m) [reply src/sidebar/lib/contextmenu/contextmenu.js]: hmm, eslint must have done those.

**Comment by nt1m** [comment]
PR #329: review- → review?(eoger)

**Comment by eoger** [comment]
Awesome work thank you!

**Comment by nt1m** [comment]
@Keith94 Nice! Note that you can make the icons white by checking "dark theme" in the settings.

**Comment by Keith94** [comment]
Ohhh that's much better!! Now I can see them properly. :-) Thanks a lot for mentioning it.

**Comment by nt1m** [comment]
@Keith94 Want to share this to [r/Firefox/](https://www.reddit.com/r/firefox/) ? I bet people there would love to see this :)
